projects/graph/graph.py

```diff
@@ -7,49 +7,198 @@ class Graph:
     """Represent a graph as a dictionary of vertices mapping labels to edges."""
     def __init__(self):
         self.vertices = {}
+        self.visited2 = set()
     def add_vertex(self, vertex):
         """
         Add a vertex to the graph.
         """
-        pass  # TODO
+        self.vertices[vertex] = set()
+        
     def add_edge(self, v1, v2):
         """
         Add a directed edge to the graph.
         """
-        pass  # TODO
+        if v1 in self.vertices and v2 in self.vertices:
+            self.vertices[v1].add(v2)
+        else:
+            raise IndexError("Can not create edge based on given vertices!")
+        
     def bft(self, starting_vertex):
         """
         Print each vertex in breadth-first order
         beginning from starting_vertex.
+
+        BFT Pseudocode
+        Create a queue
+        Create list of visited nodes
+        Put starting node in the queue
+        While: queue not empty
+        Pop first node out of queue
+        If not visited
+        Mark as visited
+        Get adjacent edges and add to list
+        Goto top of loop
+
         """
+        que = Queue()
+        visited = set()
+        que.enqueue(starting_vertex)
+        while que.size() > 0:
+            vertex = que.dequeue()
+            if vertex not in visited:
+                visited.add(vertex)
+                print(vertex)
+                for next_vert in self.vertices[vertex]:
+                    que.enqueue(next_vert)
+
         pass  # TODO
     def dft(self, starting_vertex):
         """
         Print each vertex in depth-first order
         beginning from starting_vertex.
+
+        Make a stack
+        Make visited list
+        Add first node to stack
+        While stack not empty:
+        Pop top item
+        if not visited:
+        Mark as visited
+        Get adjacent and add to stack
+
         """
-        pass  # TODO
+        questack = Stack()
+        visited = set()
+        questack.push(starting_vertex)
+        while questack.size() > 0:
+            vertex = questack.pop()
+            if vertex not in visited:
+                visited.add(vertex)
+                print(vertex)
+                for next_vert in self.vertices[vertex]:
+                    questack.push(next_vert)
+                    #print(f'{next_vert} aaaaAAAA')
+                    #print(visited)
+                    #print(self.vertices)
+                    #print(len(self.vertices))
+                    #print(len(visited))
+                    
+    
     def dft_recursive(self, starting_vertex):
         """
         Print each vertex in depth-first order
         beginning from starting_vertex.
         This should be done using recursion.
         """
-        pass  # TODO
+        if len(self.visited2) == len(self.vertices):
```

Comment:
you can add another parameter and avoid modifying a `self` property. Setting a default value for that parameter is one method of initializing that parameter the first time the function is called since that parameter (visited) wouldn't exist the first time you call it.